airbyte-integrations/connectors/source-shopify/source_shopify/shopify_graphql/bulk/job.py

```diff
@@ -498,8 +521,21 @@ def get_adjusted_job_start(self, slice_start: datetime) -> datetime:
     def _adjust_slice_end(self, slice_end: datetime, checkpointed_cursor: Optional[str] = None) -> datetime:
         """
         Choose between the existing `slice_end` value or `checkpointed_cursor` value, if provided.
+
+        Optionally: raises the `transient` error if the checkpoint collision occurs.
         """
-        return pdm.parse(checkpointed_cursor) if checkpointed_cursor else slice_end
+
+        if checkpointed_cursor:
+            self._set_last_checkpoint_cursor_value(checkpointed_cursor)
+
+            if self._checkpoint_cursor_has_collision(checkpointed_cursor):
+                raise ShopifyBulkExceptions.BulkJobCheckpointCollisionError(
+                    f"The stream: `{self.http_client.name}` checkpoint collision is detected. Try to increase the `BULK Job checkpoint (rows collected)` to the bigger value. The stream will be synced again during the next sync attempt."
+                )
```

Comment:
Is not this setting `checkpointed_cursor` to `self._job_last_checkpoint_cursor_value` and later checking If same `checkpointed_cursor` value is equal to `self._job_last_checkpoint_cursor_value`?

Reply:
Thanks. Corrected!